smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -65,6 +70,131 @@ void writeMembers(TypeScriptWriter writer, Shape shape) {
+                        () -> {
+                            MemberShape nestedMapMember = ((MapShape) memberShape).getValue();
+                            writeFilterSensitiveLogForMap(writer, nestedMapMember);
+                        }
+                    );
+                } else {
+                    // This path will never reach because of recursive isIterationRequired
+                    // adding it to not break the code, if it does reach in future
+                    writer.write("acc[key] = value;");
+                }
+                writer.write("return acc;");
+            }
+        );
+    }
+
+    void writeFilterSensitiveLog(TypeScriptWriter writer, Shape shape) {
+        writer.write("...obj,");
+        for (MemberShape member : members) {
+            Shape memberShape = model.expectShape(member.getTarget());
+            String memberName = TypeScriptUtils.sanitizePropertyName(symbolProvider.toMemberName(member));
+            if (member.getMemberTrait(model, SensitiveTrait.class).isPresent()) {
+                // member is Sensitive, hide the value
+                writer.write("...(obj.${L} && { ${L}: SENSITIVE_STRING }),", memberName, memberName);
+            } else if (memberShape instanceof StructureShape) {
+                // Call filterSensitiveLog on Structure
+                writer.write("...(obj.${L} && { ${L}: ${T}.filterSensitiveLog(obj.${L})}),",
+                    memberName, memberName, symbolProvider.toSymbol(member), memberName);
+            } else if (memberShape instanceof CollectionShape) {
+                MemberShape collectionMember = ((CollectionShape) memberShape).getMember();
+                if (isIterationRequired(model.expectShape(collectionMember.getTarget()))) {
+                    // Iterate over array items, and call array specific function on each member
+                    writer.openBlock("...(obj.${L} && { ${L}: obj.${L}.map(", ")}),",
+                        memberName, memberName, memberName,
+                        () -> {
+                            writeFilterSensitiveLogForCollection(writer, collectionMember);
+                        }
+                    );
+                }
+            } else if (memberShape instanceof MapShape) {
+                MemberShape mapMember = ((MapShape) memberShape).getValue();
+                if (isIterationRequired(model.expectShape(mapMember.getTarget()))) {
+                    // Iterate over Object entries, and call reduce to repopulate map
+                    writer.openBlock("...(obj.${L} && { ${L}: Object.entries(obj.${L}).reduce(", ")}),",
+                        memberName, memberName, memberName,
+                        () -> {
+                            writeFilterSensitiveLogForMap(writer, mapMember);
+                        }
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Identified if iteration is required on shape
```

Comment:
Make sure this gets built successfully locally, javadoc will complain if sentences don't end with periods. There may be other issues if it hasn't run successfully either.

Reply:
Added period for all comments to follow the standard in commit https://github.com/awslabs/smithy-typescript/pull/170/commits/1584a2a8896b443705554539971fdb18eb4c6831